internal/dispatch/remote/cluster.go

```diff
@@ -31,6 +38,11 @@ func (cr *clusterDispatcher) DispatchCheck(ctx context.Context, req *v1.Dispatch
 		return &v1.DispatchCheckResponse{Metadata: emptyMetadata}, err
 	}
 
+	requestKey, err := protoMarshal.Marshal(req)
```

Comment:
I think we should manually be creating the request keys, not just using the entire dispatched proto.

Reply:
Could you double check the fields I included? I left out depth and limit but didn't fully convince myself that was okay.

Reply:
I didn't realize the current impl just marshall's the protos, I wonder if that's contributing to our hit rate being low: https://github.com/authzed/spicedb/blob/e5ebe46d37a6a23965f97042161df979361855ab/internal/dispatch/client/consistentbackend/client.go#L200